src/test/shell/bazel/bazel_determinism_test.sh

```diff
@@ -69,18 +69,6 @@ function test_determinism()  {
     # Set up the maven repository properly.
     cp derived/maven/BUILD.vendor derived/maven/BUILD
 
-    # Update the hash of bazel_tools in lockfile to avoid rerunning module resolution.
```

Comment:
I think that you can also delete the backwards compatiblity logic in `RepoSpec` and `BzlmodRepoRuleFunction`.

Reply:
will send another PR for this.